projectile.el

```diff
@@ -1299,7 +1308,9 @@ explicitly."
 (defun projectile-default-project-name (project-root)
   "Default function used to create the project name.
 The project name is based on the value of PROJECT-ROOT."
-  (file-name-nondirectory (directory-file-name project-root)))
+  (or (let ((default-directory project-root))
+        (nth 3 (projectile-parse-dirconfig-file)))
```

Comment:
As the project name is constantly recalculated in the modeline parsing the config file all the time can make things extremely slow. That's one more reason I'm wary about the proposed change.